src/Workspaces/Core/Portable/Workspace/Solution/SolutionInfo.cs

```diff
@@ -139,7 +141,7 @@ public static SolutionAttributes ReadFrom(ObjectReader reader)
                 return new SolutionAttributes(solutionId, VersionStamp.Create(), filePath);
             }
 
-            private Checksum _lazyChecksum;
+            private Checksum? _lazyChecksum;
```

Comment:
Is there a reason we opted to use null here instead of `Lazy<Checksum>`?

Reply:
We use this pattern in other places, I suspect to avoid the extra allocation overhead. @heejaechang?